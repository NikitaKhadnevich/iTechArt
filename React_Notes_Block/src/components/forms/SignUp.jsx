/* eslint-disable react/prop-types */
/* eslint-disable react/jsx-no-bind */
/* eslint-disable no-unused-vars */
import * as React from 'react';
import Avatar from '@mui/material/Avatar';
import TextField from '@mui/material/TextField';
import Button from '@material-ui/core/Button';
import Grid from '@mui/material/Grid';
import LockOutlinedIcon from '@mui/icons-material/LockOutlined';
import PropTypes from 'prop-types';
import Typography from '@mui/material/Typography';
import {
  SignContainer,
  SignBox,
  FormBox,
  FormGrid,
  FormButton,
} from './styled';

export default function SignUp({ formik }) {
  const {
    values: { email, firstName, lastName, birthday, password, confirmPassword },
    errors,
    touched,
    handleChange,
    isValid,
    setFieldTouched,
    handleBlur,
    handleSubmit,
  } = formik;

  function change(name, e) {
    handleChange(e);
    setFieldTouched(name, true, false);
  }

  return (
    <SignContainer component='main'>
      <SignBox>
        <Avatar sx={{ m: 1, bgcolor: 'secondary.main' }}>
          <LockOutlinedIcon />
        </Avatar>
        <Typography component='h5' variant='h5'>
          Sign up
        </Typography>
        <FormBox
          component='form'
          noValidate
          sx={{ mt: 3 }}
          onSubmit={handleSubmit}
        >
          <FormGrid container spacing={2}>
            <Grid item xs={12}>
              <TextField
                value={email}
                id='email'
                name='email'
                required
                variant='filled'
                autoFocus
                fullWidth
                label='Email Address'
                autoComplete='email'
                error={touched.email && Boolean(errors.email)}
                helperText={touched.email ? errors.email : ''}
                onChange={change.bind(null, 'email')}
                onBlur={handleBlur}
              />
            </Grid>
            <Grid item xs={12} sm={6}>
              <TextField
                value={firstName}
                id='firstName'
                name='firstName'
                required
                fullWidth
                label='First Name'
                autoComplete='given-name'
                error={touched.firstName && Boolean(errors.firstName)}
                helperText={touched.firstName ? errors.firstName : ''}
                onChange={change.bind(null, 'firstName')}
                onBlur={handleBlur}
              />
            </Grid>
            <Grid item xs={12} sm={6}>
              <TextField
                value={lastName}
                id='lastName'
                name='lastName'
                required
                fullWidth
                label='Last Name'
                autoComplete='given-name'
                error={touched.lastName && Boolean(errors.lastName)}
                helperText={touched.lastName ? errors.lastName : ''}
                onChange={change.bind(null, 'lastName')}
                onBlur={handleBlur}
              />
            </Grid>
            <Grid item xs={12}>
              <TextField
                value={birthday}
                id='birthday'
                name='birthday'
                required
                fullWidth
                label='Date of Birth'
                error={touched.birthday && Boolean(errors.birthday)}
                helperText={touched.birthday ? errors.birthday : ''}
                onChange={change.bind(null, 'birthday')}
                onBlur={handleBlur}
              />
            </Grid>
            <Grid item xs={12}>
              <TextField
                id='password'
                name='password'
                required
                fullWidth
                label='Password'
                type='password'
                autoComplete='new-password'
                error={touched.password && Boolean(errors.password)}
                helperText={touched.password ? errors.password : ''}
                onChange={change.bind(null, 'password')}
                onBlur={handleBlur}
              />
            </Grid>
            <Grid item xs={12}>
              <TextField
                id='confirmPassword'
                name='confirmPassword'
                required
                fullWidth
                label='Confirm Password'
                type='password'
                error={
                  touched.confirmPassword && Boolean(errors.confirmPassword)
                }
                helperText={
                  touched.confirmPassword ? errors.confirmPassword : ''
                }
                onChange={change.bind(null, 'confirmPassword')}
                onBlur={handleBlur}
              />
            </Grid>
            <Grid item xs={12}>
              <Button
                disabled={!isValid}
                type='submit'
                fullWidth
                color='secondary'
                variant='outlined'
              >
                Sign Up
              </Button>
            </Grid>
          </FormGrid>
        </FormBox>
      </SignBox>
    </SignContainer>
  );
}

SignUp.propTypes = {
  // email: PropTypes.string,
  // confirmPassword: PropTypes.string,
  // errors: PropTypes.bool,
  // touched: PropTypes.bool,
  // handleSubmit: PropTypes.func,
  // handleChange: PropTypes.func,
  // isValid: PropTypes.bool,
  // setFieldTouched: PropTypes.func,
};

SignUp.defaultProps = {
  // email: 'values',
  // confirmPassword: 'confirmPassword',
  // errors: 'errors',
  // touched: 'touched',
  // handleSubmit: 'handleSubmit',
  // handleChange: 'handleChange',
  // isValid: 'isValid',
  // setFieldTouched: 'setFieldTouched'
};
