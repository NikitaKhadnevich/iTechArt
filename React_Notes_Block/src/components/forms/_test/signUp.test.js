/* eslint-disable prefer-const */
/* eslint-disable no-unused-vars */
/* eslint-disable no-shadow */
/* eslint-disable react/jsx-props-no-spreading */
import React, { useEffect } from 'react';
import * as reactRedux from 'react-redux';
import { configure, shallow, mount } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import SignUp from '../SignUp';

configure({ adapter: new Adapter() }); // ES6 modules

describe('SignUp', () => {
  const formik = {
    errors: false,
    touched: false,
    handleChange: jest.fn(),
    isValid: false,
    setFieldTouched: jest.fn(),
    handleBlur: jest.fn(),
    handleSubmit: jest.fn(),
    values: {
      email: '',
      firstName: '',
      lastName: '',
      birthday: '',
      password: '',
      confirmPassword: '',
      sharedNotes: [],
    },
  };
  const signUpLoaded = {
    formik,
    isFetching: false,
    isLoading: false,
    hasAlready: 'some Text',
  };
  const signUpLoading = {
    formik,
    isFetching: true,
    isLoading: true,
    hasAlready: 'some Text',
  };

  const useSelectorMock = jest.spyOn(reactRedux, 'useSelector');
  const useDispatchMock = jest.spyOn(reactRedux, 'useDispatch');

  beforeEach(() => {
    useSelectorMock.mockClear();
    useDispatchMock.mockClear();
  });

  it('Is it SignUp Exist and Loaded ', () => {
    const signInWrapper = shallow(<SignUp {...signUpLoaded} />);
    expect(signInWrapper).toMatchSnapshot();
    expect(signInWrapper).toHaveLength(1);
  });

  it('should components data are loading/Fetching', () => {
    const loadingWrapper = shallow(<SignUp {...signUpLoading} />);
    expect(loadingWrapper.find('Spinner')).toHaveLength(1);
    expect(loadingWrapper).toMatchSnapshot();
  });
});
