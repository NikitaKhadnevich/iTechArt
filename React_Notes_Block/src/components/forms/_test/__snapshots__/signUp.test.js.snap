// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp Is it SignUp Exist and Loaded  1`] = `
<Styled(ForwardRef(Container))
  component="main"
>
  <Styled(ForwardRef(Box))>
    <SignUpSuccess />
    <Styled(ForwardRef(Box))
      component="form"
      noValidate={true}
      onSubmit={[MockFunction]}
      sx={
        Object {
          "mt": 3,
        }
      }
    >
      <SignUpGrid
        errors={false}
        handleBlur={[MockFunction]}
        isValid={false}
        onChangeAccum={[Function]}
        touched={false}
        values={
          Object {
            "birthday": "",
            "confirmPassword": "",
            "email": "",
            "firstName": "",
            "lastName": "",
            "password": "",
            "sharedNotes": Array [],
          }
        }
      />
    </Styled(ForwardRef(Box))>
  </Styled(ForwardRef(Box))>
</Styled(ForwardRef(Container))>
`;

exports[`SignUp should components data are loading/Fetching 1`] = `
<Styled(ForwardRef(Container))
  component="main"
>
  <Spinner />
</Styled(ForwardRef(Container))>
`;
